src/main/java/com/nutomic/syncthingandroid/activities/ShareToFolderActivity.java

```diff
@@ -0,0 +1,52 @@
+package com.nutomic.syncthingandroid.activities;
+
+import android.content.ComponentName;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.IBinder;
+import android.os.Bundle;
+
+import com.nutomic.syncthingandroid.R;
+import com.nutomic.syncthingandroid.fragments.SelectFolderFragment;
+
+import java.util.ArrayList;
+
+public class ShareToFolderActivity extends SyncthingActivity {
+
+    private SelectFolderFragment mSelectFolderFragment;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_share_to_folder);
+        mSelectFolderFragment = (SelectFolderFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.share_to_folder_fragment);
+        mSelectFolderFragment .setHasOptionsMenu(false);
+
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        String type = intent.getType();
+
+        ArrayList<Uri> uris = new ArrayList<>();
+        if (Intent.ACTION_SEND.equals(action) && type != null) {
```

Comment:
Why are you checking the type? And if that's really needed, you should do something (at least logging) if it is null.